### PR TITLE
Upload coverage action

### PR DESCRIPTION
**Pull request by AivarasGluodas**

### CODE REVIEW

File: actions/upload-coverage/src/util/json-client/json-client.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+export const readJsonWithComments = (path: string): any => {
+  return JSON.parse(
+    fs
+      .readFileSync(path, 'utf-8')
+      .replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g, (jsonKeyOrValue, comment) =>
+        comment ? '' : jsonKeyOrValue
+      )
+  );
+};
```
Comment (SSinkus) [inline actions/upload-coverage/src/util/json-client/json-client.ts:11]: I saw this more than once in separate actions. Maybe we should start thinking about `common-utils` package? @raihle @AivarasGluodas @ech018

File: actions/upload-coverage/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# get-preview-env-config
```
Comment (ech018) [inline actions/upload-coverage/README.md:1]: Wrong readme.

**Review by ech018 — APPROVED** [review]

File: actions/upload-coverage/src/util/json-client/json-client.ts (hunk 1 above)
Comment (AivarasGluodas) [reply actions/upload-coverage/src/util/json-client/json-client.ts:11]: yeah - it's something i tried to bring up today. I also added this to our conventions agenda

File: actions/upload-coverage/src/lib/create-coverage-artifact/create-coverage-artifact.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  const coverageReports = rushPackages.map((pkg) =>
+    getCoverageForPackage({
+      name: pkg.packageName,
+      path: pkg.projectFolder,
+    })
+  );
+  const totalCoverage = calculateTotalCoverage(coverageReports);
+
+  writefile(COVERAGE_FILE_NAME, JSON.stringify(totalCoverage, null,2));
+
```
Comment (SSinkus) [inline actions/upload-coverage/src/lib/create-coverage-artifact/create-coverage-artifact.ts:19]: Why we can't use just writeFileSync?

Comment (AivarasGluodas) [reply actions/upload-coverage/src/lib/create-coverage-artifact/create-coverage-artifact.ts:19]: we are using writeFileSync this is just a wrapper around it

File: actions/upload-coverage/README.md (hunk 2 above)
Comment (AivarasGluodas) [reply actions/upload-coverage/README.md]: updated

**Pull request merged**